docs/contributing/setting-up-project.mdx

```diff
@@ -2,36 +2,80 @@
 title: Setup Guide
 ---
 
-#### 1. Automated dependency installation
+## 1. Automated dependency installation
 Installing all the dependencies
 ```bash
  make deps
 ```
 
-#### 2. Build
-To build and rebuild after changes to Rust files
+## 2. Build
+
+To build and rebuild after changes to Rust files.
+
+<Note>
+Make sure you have the Rust compiler installed. This package requires Rust and Cargo to compile extensions.
+</Note>
+
+### Install Dependencies
 
-Note: Make sure you have rust compiler installed. This package requires Rust and Cargo to compile extensions.
 ```bash
- make install
+make install
 ```
 
-#### 3. Test
+### Troubleshooting
+
+#### Issue: `python: command not found`
+
+On some systems, the `make install` command may fail with an error stating that the `python` command is not found. This happens because some systems only provide `python3` and do not include a `python` symlink.
+
+#### macOS Fix:
+If you're on macOS, create a symbolic link for `python` pointing to `python3`:
+
+```bash
+sudo ln -s $(which python3) /usr/local/bin/python
+```
+
+#### Linux Fix:
+
+```bash
+sudo ln -s $(which python3) /usr/bin/python
+```
+
+#### Windows Fix:
+On Windows, ensure Python is installed and added to your `PATH`. You can check by running:
+
+```powershell
+python --version
+```
+
+If `python` is not recognized, use the `python3` command instead, or create an alias in PowerShell:
+
+```powershell
+Set-Alias -Name python -Value python3
+```
+
+After applying the appropriate fix, retry:
+
+```bash
+make install
+```
```

Comment:
This section should actually be next to `make deps`, and should refer to `make deps` instead of `make install`. From looking at the Makefile, that command is the one that actually invokes `python` directly.